docs/java-api-quickstart.md

```diff
@@ -38,37 +38,42 @@ The Hive catalog connects to a Hive metastore to keep track of Iceberg tables.
 You can initialize a Hive catalog with a name and some properties.
 (see: [Catalog properties](../configuration/#catalog-properties))
 
-**Note:** Currently, `setConf` is always required for hive catalogs, but this will change in the future.
+### Using a Hive catalog
+
+The Hive catalog connects to a Hive metastore to keep track of Iceberg tables.
+You can initialize a Hive catalog with a name and some properties.
+(see: [Catalog properties](../configuration/#catalog-properties))
 
 ```java
+import java.util.HashMap
+import java.util.Map
+
 import org.apache.iceberg.hive.HiveCatalog;
 
 HiveCatalog catalog = new HiveCatalog();
-catalog.setConf(spark.sparkContext().hadoopConfiguration());  // Configure using Spark's Hadoop configuration
```

Comment:
I think this should be included, but noted as optional. If the caller has a Hadoop conf, they should supply it. And the caller will almost certainly have one if they are using the Hive Metastore.